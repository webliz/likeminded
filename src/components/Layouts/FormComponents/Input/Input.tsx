/** React */
import React from 'react';

interface InputField {
  id?: string,
  type: string, 
  name: string, 
  value: string, 
  placeholder?: string,
  disabled?: boolean, 
  className?: string, 
  style?: React.CSSProperties, 
  max?: number, 
  min?: number,
  onChange: any
}

/** Form input field  */
const Input:React.FC<InputField> 
  = ({id, type, name, value, placeholder, onChange, disabled, className, style, max, min}) => {
  return (
    <input 
      id={id}
      type={type} 
      name={name} 
      value={value} 
      placeholder={placeholder} 
      disabled={disabled} 
      className={className} 
      style={style} 
      maxLength={max} 
      minLength={min} 
      onChange={onChange}
      data-testid='form-input'
    />
  )
}

export default Input;