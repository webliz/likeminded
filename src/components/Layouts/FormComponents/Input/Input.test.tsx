import { render, screen } from '@testing-library/react';
import Input from './Input';
/**  Snapshots  rendering */
import renderer from 'react-test-renderer';

describe('Input component', () => {

  test('renders correctly', () => {
    const component = renderer.create(<Input type='password' name='password' value='' onChange={(e: any) => console.log(e)}/>).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render the component on the screen', () => {
    render(<Input data-testid='form-input' type='password' name='password' value='' onChange={(e: any) => console.log(e)}/>);
    expect(screen.getByTestId('form-input')).toBeInTheDocument();
  });

});