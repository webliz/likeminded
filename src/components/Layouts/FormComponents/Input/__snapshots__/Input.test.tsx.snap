// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<input
  data-testid="form-input"
  name="password"
  onChange={[Function]}
  type="password"
  value=""
/>
`;
